### PR TITLE
fix: honor vertical scale prop in 3D

### PR DESCRIPTION
**Pull request by hkfb**
Note that only the first `verticalScale` value is taken into account. To handle viewport dependent scaling we can look into creating a custom view following [this discussion](https://github.com/visgl/deck.gl/discussions/8675)

### CODE REVIEW

File: typescript/packages/subsurface-viewer/src/components/Map.tsx [hunk 1]
```diff
@@ -504,8 +424,20 @@ const Map: React.FC<MapProps> = ({
...
+    const { zScale: zReScale, divRef: zScaleRef } = useHandleRescale();
+
+    const { shiftHeld, divRef: shiftHeldRef } = useShiftHeld();
+
+    const divRef = mergeRefs(
+        zScaleRef,
+        shiftHeldRef
+    ) as React.Ref<HTMLDivElement>;
+
+    const zScale = verticalScale ?? zReScale;
```
Comment (t0oF-azpn) [inline typescript/packages/subsurface-viewer/src/components/Map.tsx:440]: Does that mean that if any viewport has a `verticalScale` set, the arrow keys are disabled ?
In that case, we probably want to disable the event handling, no ?

Does this allow the client application to set its own handler of key events ?

File: typescript/packages/subsurface-viewer/src/views/viewport.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    rotationX?: number;
+    rotationOrbit?: number;
+
+    /** A vertical scale factor, used to scale items in the viewport vertically */
+    verticalScale?: number;
+
+    isSync?: boolean;
+}
+
+export const useVerticalScale = (viewport: ViewportType[] | undefined) => {
```
Comment (t0oF-azpn) [inline typescript/packages/subsurface-viewer/src/views/viewport.ts:38]: Please, rename `viewport` to `viewports`.

So here we use the first found verticalScale from all the viewports, right ?

**Review by t0oF-azpn — CHANGES_REQUESTED** [review]

File: typescript/packages/subsurface-viewer/src/components/Map.tsx (hunk 1 above)
Comment (hkfb) [reply typescript/packages/subsurface-viewer/src/components/Map.tsx:440]: > Does that mean that if any viewport has a `verticalScale` set, the arrow keys are disabled ? In that case, we probably want to disable the event handling, no ?

Yes, the arrow keys will be disabled. Will add a change to disable the events.

> 
> Does this allow the client application to set its own handler of key events ?

That was the plan initially, to add let `verticalScale` be something like `number | () => number` and, if not set, fall back to a default value like the current event handler hook. So probably in an upcoming PR.

File: typescript/packages/subsurface-viewer/src/views/viewport.ts (hunk 2 above)
Comment (hkfb) [reply typescript/packages/subsurface-viewer/src/views/viewport.ts]: > Please, rename `viewport` to `viewports`.

Done.
> 
> So here we use the first found verticalScale from all the viewports, right ?

Yes, will make an attempt to make `verticalScale` viewport dependent, like in 2D, in a subsequent PR, and then we can apply a `verticalScale` respectively to each viewport.

**Review by t0oF-azpn — APPROVED** [review]

**Pull request merged**